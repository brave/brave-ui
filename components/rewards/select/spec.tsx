/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Select from './index'

describe('Select tests', () => {
  const baseComponent = (props?: object) => <Select id='select' onChange={() => false}
    >
      <div data-value='0'>01</div>
      <div data-value='1'>11</div>
      <div data-value='2'>12</div>
    </Select>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#select').length
      expect(assertion).toBe(1)
    })
  })
})
