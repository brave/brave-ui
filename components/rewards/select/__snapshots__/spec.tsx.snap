// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select tests basic tests matches the snapshot 1`] = `
.c0 {
  max-width: 254px;
  width: 100%;
}

.c1 {
  width: 100%;
  font-family: Poppins;
  line-height: normal;
  font-size: 14px;
  font-weight: 500;
  color: #686978;
  margin-bottom: 6px;
}

.c2 {
  position: relative;
  outline: 0;
}

.c3 {
  line-height: normal;
  border-radius: 3px;
  border: solid 1px #dfdfe8;
  font-family: Poppins;
  font-size: 14px;
  color: #686978;
  background: #fff;
  width: 100%;
  padding: 10px 31px 10px 13px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  background: url(test-file-stub) no-repeat right 15px top 16px #fff;
}

.c4 {
  display: none;
  position: absolute;
  top: calc(100% + 4px);
  left: 0;
  width: 100%;
  border-radius: 3px;
  box-shadow: 0 2px 5px 0 rgba(223,223,232,0.5);
  background-color: #ffffff;
  border: solid 1px #dfdfe8;
  overflow: hidden;
}

.c5 {
  font-family: Poppins;
  font-size: 14px;
  line-height: 2.57;
  color: #1b1d2f;
  position: relative;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  padding: 1px 21px 0 30px;
  background: url(test-file-stub) no-repeat 14px 12px #e9f0ff;
}

.c6 {
  font-family: Poppins;
  font-size: 14px;
  line-height: 2.57;
  color: #1b1d2f;
  position: relative;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  padding: 1px 21px 0 30px;
}

<div
  className="c0"
  id="select"
>
  <div
    className="c1"
  />
  <div
    className="c2"
    onBlur={[Function]}
    tabIndex="0"
  >
    <div
      className="c3"
      disabled={undefined}
      onClick={[Function]}
    >
      01
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
        onClick={[Function]}
        selected={true}
      >
        01
      </div>
      <div
        className="c6"
        onClick={[Function]}
        selected={false}
      >
        11
      </div>
      <div
        className="c6"
        onClick={[Function]}
        selected={false}
      >
        12
      </div>
    </div>
  </div>
</div>
`;
